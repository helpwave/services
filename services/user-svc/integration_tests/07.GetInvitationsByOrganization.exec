#!/bin/bash

echo "exists with state:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"organization_id": "cc401e43-d76c-4f0e-933a-3ddfab895e59", "state": 2}' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/GetInvitationsByOrganization

echo "exists without state:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"organization_id": "cc401e43-d76c-4f0e-933a-3ddfab895e59"}' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/GetInvitationsByOrganization
