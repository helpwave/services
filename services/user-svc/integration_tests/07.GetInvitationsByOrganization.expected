exists with state:
{
  "invitations": [
    {
      "id": "dcf6280b-dee6-4a0f-826b-cc506476e1e9",
      "email": "invitation@mail.de",
      "organizationId": "cc401e43-d76c-4f0e-933a-3ddfab895e59",
      "state": "INVITATION_STATE_ACCEPTED"
    },
    {
      "id": "d0bd85ee-910a-4578-bbdc-a4346a624a34",
      "email": "mailer@helpwave.de",
      "organizationId": "cc401e43-d76c-4f0e-933a-3ddfab895e59",
      "state": "INVITATION_STATE_ACCEPTED"
    },
    {
      "id": "07c4f3fd-5146-4bda-aece-c02aa0c50fdc",
      "email": "marvin@helpwave.de",
      "organizationId": "cc401e43-d76c-4f0e-933a-3ddfab895e59",
      "state": "INVITATION_STATE_ACCEPTED"
    }
  ]
}
exists without state:
{
  "invitations": [
    {
      "id": "dcf6280b-dee6-4a0f-826b-cc506476e1e9",
      "email": "invitation@mail.de",
      "organizationId": "cc401e43-d76c-4f0e-933a-3ddfab895e59",
      "state": "INVITATION_STATE_ACCEPTED"
    },
    {
      "id": "d0bd85ee-910a-4578-bbdc-a4346a624a34",
      "email": "mailer@helpwave.de",
      "organizationId": "cc401e43-d76c-4f0e-933a-3ddfab895e59",
      "state": "INVITATION_STATE_ACCEPTED"
    },
    {
      "id": "07c4f3fd-5146-4bda-aece-c02aa0c50fdc",
      "email": "marvin@helpwave.de",
      "organizationId": "cc401e43-d76c-4f0e-933a-3ddfab895e59",
      "state": "INVITATION_STATE_ACCEPTED"
    },
    {
      "id": "4e857c97-0afc-4e3e-97a6-cc9d346f68aa",
      "email": "tom.breitband@helpwave.de",
      "organizationId": "cc401e43-d76c-4f0e-933a-3ddfab895e59",
      "state": "INVITATION_STATE_PENDING"
    }
  ]
}
